File: bounce/src/test/java/is/ricko/bounce/ABounceOfflineTests.java
package is.ricko.bounce;

import org.mockito.Mockito;

import java.util.Objects;
import java.util.UUID;

public abstract class ABounceOfflineTests {
  protected <T> T anyNull() {
    Mockito.any();
    return uninitialized();
  }

  protected <T> T nonNull(final T t) {
    Mockito.eq(t);
    return t;
  }

  protected int randomInt() {
    return Objects.hashCode(Math.random());
  }

  protected String randomString() {
    return UUID.randomUUID().toString();
  }

  protected <T> T uninitialized() {
    return null;
  }
}

File: bounce/src/test/kotlin/is/ricko/bounce/ABounceOfflineTests.kt
package `is`.ricko.bounce

import org.mockito.Mockito
import java.util.*

abstract class ABounceOfflineTests {
    fun randomString(): String {
        return UUID.randomUUID().toString()
    }

    fun randomInt(): Int {
        return Math.random().toInt()
    }

    fun <T> nonNull(t: T): T {
        Mockito.eq(t)
        return t
    }

    fun <T> anyNull(): T {
        Mockito.any<T>()
        return uninitialized()
    }

    @Suppress("UNCHECKED_CAST")
    private fun <T> uninitialized(): T = null as T
}